docs/user/exercises/test-case-best-practices.inc

```diff
@@ -0,0 +1,406 @@
+     * Problem Statement Task 2.1
+     * Worth 1.5 Points (Weight: 1)
+     */
+    @Test
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+  Better yet, for manual correction, use these comments in the display name of the test. This allows the assessors, who execute the
+  tests in the IDE, to have more meaningful names displayed. The following example would make counting points easier.
+
+  .. code-block:: java
+
+    @DisplayName("1.5 P | Books can be borrowed successfully")
+    @Test
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+- **Use appropriate and descriptive names for test cases**
+
+  After exercises and exams, test names will be used to create statistics. If the tests are called *test1*, *test2*, *test3*,
+  it will be hard to read those statistics. This is the same reason as to why you should not name your variables ``int a``,
+  ``double b``, ``String c``. For example, if you want to test the method ``borrow`` in the class ``Book``,
+  ``testBorrowInBook()`` would be an appropriate name for the test case.
+
+  .. code-block:: java
+
+    @Test
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+  If you have many tests in different (nested) classes that are not completely distinct, add the name of the tested
+  class to the test to avoid having two tests with the same names. For example, if you test both ``add`` methods of a
+  LinkedList and an ArrayList. Clear and distinct test names also makes it easier to read and configure the grading in Artemis.
+
+  .. code-block:: java
+
+    @Test
+    public void test_LinkedList_add() {
+        // Test Code
+    }
+
+  .. hint::
+    For Java exercises: If all test methods are in a single class this is not necessary, because the Java compiler
+    won't allow multiple methods with override-equivalent signatures.
+
+- **Use appropriate timeouts for test cases**
+
+  For regular test cases, using a ``@StrictTimeout(1)``  annotation is enough. This represents a strict timeout of one
+  second. The value type of the strict timeout annotation defaults to seconds. If you need a shorter timeout, you can use
+  ``@StrictTimeout(value = 500, unit = TimeUnit.MILLISECONDS)``.
+
+  .. code-block:: java
+
+    @Test
+    @StrictTimeout(1)
+    public void testBorrowInBook() {
+        // Test Code
+    }
+
+  .. note::
+      When defining timeouts, you should take into account that the tests are run on a Continuous Integration Server
+      (using build agents). The tests will most likely execute a lot faster on your local machine.
+
+- **Avoiding assert statements**
+
+  Instead use conditional ``fail()`` calls to hide confusing information from the students. This could be considered a
+  bad practice in regular testing but it helps to create fail messages that are less confusing, especially for beginners.
+  Additionally, this also hides information about test implementation details.
```

Comment:
This should only be done when the used and expected values should be hidden or the students wouldn't understand the output of JUnit.
So especially for hidden tests they may still be an option